hopefully address:
http://bugs.gentoo.org/253570
http://savannah.gnu.org/bugs/?25294

upstream commit e3bcac430a07a05ba2ac66587d168a842ba8c0ea which was done for
unrelated reasons ...

diff --git a/find/ftsfind.c b/find/ftsfind.c
index 4a33059..765221b 100644
--- a/find/ftsfind.c
+++ b/find/ftsfind.c
@@ -472,7 +472,7 @@ consider_visiting(FTS *p, FTSENT *ent)
       || ent->fts_info == FTS_NS /* e.g. symlink loop */)
     {
       assert (!state.have_stat);
-      assert (state.type != 0);
+      assert (ent->fts_info == FTS_NSOK || state.type != 0);
       mode = state.type;
     }
   else
